python/pyspark/pandas/tests/test_typedef.py

```diff
@@ -56,6 +56,23 @@
 
 
 class TypeHintTests(unittest.TestCase):
+    def test_infer_schema_with_no_return(self):
+        def try_infer_return_type():
+            def f():
+                pass
+
+            infer_return_type(f)
+
+        self.assertRaisesRegex(TypeError, "Unsupported callable", try_infer_return_type)
```

Comment:
Can we refine the message bit more explicitly ??

e.g. "A return value is required for the function passed to `infer_return_type`."

or something.

Reply:
updated. Thanks!